src/solvers/mp/MasterProblem.cpp

```diff
@@ -0,0 +1,1278 @@
+				sprintf(name, "minWorkedDaysAvgVar_N%d", i);
+				pModel_->createPositiveVar(&minWorkedDaysAvgVars_[i], name, weightTotalShiftsAvg_[i]);
+
+				sprintf(name, "minWorkedDaysAvgCons_N%d", i);
+				vector<MyVar*> varsAvg1 = {minWorkedDaysVars_[i], minWorkedDaysAvgVars_[i]};
+				vector<double> coeffsAvg1 = {1,1};
+				pModel_->createGEConsLinear(&minWorkedDaysAvgCons_[i], name, minTotalShiftsAvg_[i], varsAvg1, coeffsAvg1);
+
+				isMinWorkedDaysAvgCons_[i] = true;
+			}
+
+			if (maxTotalShiftsAvg_[i] < maxTotalShifts_[i]) {
+				sprintf(name, "maxWorkedDaysAvgVar_N%d", i);
+				pModel_->createPositiveVar(&maxWorkedDaysAvgVars_[i], name, weightTotalShiftsAvg_[i]);
+
+				sprintf(name, "maxWorkedDaysAvgCons_N%d", i);
+				vector<MyVar*> varsAvg2 = {maxWorkedDaysVars_[i],maxWorkedDaysAvgVars_[i]};
+				vector<double> coeffsAvg2 = {-1,-1};
+				pModel_->createLEConsLinear(&maxWorkedDaysAvgCons_[i], name, maxTotalShiftsAvg_[i], varsAvg2, coeffsAvg2);
+
+				isMaxWorkedDaysAvgCons_[i] = true;
+			}
+		}
+
+		sprintf(name, "maxWorkedWeekendVar_N%d", i);
+		pModel_->createPositiveVar(&maxWorkedWeekendVars_[i], name, weightTotalWeekendsMax_[i]);
+
+		sprintf(name, "maxWorkedWeekendCons_N%d", i);
+		vector<MyVar*> vars3 = {maxWorkedWeekendVars_[i]};
+		vector<double> coeffs3 = {-1};
+		pModel_->createLEConsLinear(&maxWorkedWeekendCons_[i], name, maxTotalWeekends_[i],
+				vars3, coeffs3);
+
+		// STAB:Add stabilization variable
+		//
+		if (param.isStabilization_) {
+			sprintf(name,"stabMaxWorkedWeekendMinus_%i",i);
+			pModel_->createPositiveVar(&stabMaxWorkedWeekendMinus_[i],name,-param.stabCostIni_+param.stabCostMargin_,DEFAULT_PATTERN,0,param.stabBoundIni_);
+			pModel_->addCoefLinear(maxWorkedWeekendCons_[i],stabMaxWorkedWeekendMinus_[i],-1.0);
+		}
+
+		// STAB: not implemented there yet
+		if ( !maxTotalWeekendsAvg_.empty()  && !weightTotalWeekendsAvg_.empty()
+				&& maxTotalWeekendsAvg_[i] < theLiveNurses_[i]->maxTotalWeekends() - theLiveNurses_[i]->pStateIni_->totalWeekendsWorked_) {
+
+			sprintf(name, "maxWorkedWeekendAvgVar_N%d", i);
+			pModel_->createPositiveVar(&maxWorkedWeekendAvgVars_[i], name, weightTotalWeekendsAvg_[i]);
+
+			sprintf(name, "maxWorkedWeekendAvgCons_N%d", i);
+			vector<MyVar*> varsAvg3 = {maxWorkedWeekendVars_[i],maxWorkedWeekendAvgVars_[i]};
+			vector<double> coeffsAvg3 = {-1,-1};
+			pModel_->createLEConsLinear(&maxWorkedWeekendAvgCons_[i], name, maxTotalWeekendsAvg_[i]- theLiveNurses_[i]->pStateIni_->totalWeekendsWorked_,
+					varsAvg3, coeffsAvg3);
+
+			isMaxWorkedWeekendAvgCons_[i] = true;
+
+		}
+
+	}
+
+	// WEEKEND CUTS
+	// sprintf(name, "sumMaxWorkedWeekendCons");
+	// std::vector<pVar> varsSum3;
+	// std::vector<double> coeffsSum3;
+	// for(int i=0; i<pScenario_->nbNurses_; i++) {
+	// 	varsSum3.push_back(maxWorkedWeekendVars_[i]);
+	// 	coeffsSum3.push_back(1);
+	// }
+	// pModel_->createGEConsLinear(&sumMaxWorkedWeekendCons_, name, 4, varsSum3, coeffsSum3);
```

Comment:
Remember what this cut is ? Delete the comment block?

Reply:
done in c4f0269a58e45935393e82d30bf0281979923c5e